packages/dashboard/src/components/contextMenu/index.test.tsx

```diff
@@ -0,0 +1,378 @@
+import * as React from 'react';
```

Comment:
These tests are only testing that a function gets called but don't actually test the functionality. From a users perspective I think we would benefit more from writing integration tests for these.

Reply:
I think integration tests would be useful, and we can certainly add those, but I disagree that these tests won't catch bugs that arise from the context menu. Also, I don't know if we would want to have to setup the entire dashboard just to ensure that the context menu has the right buttons and that those buttons are clickable under the right circumstances. I think a test like that would end up failing more because we changed the api of the dashboard and didn't update our test setup accordingly, than failing because we actually made a breaking change to the context menu.

I think what might be a better solution though is to refactor the context menu component to use the redux actions directly like I did with the keyboard shortcuts instead of passing them in as a function. Then I can move the jest mocks onto those redux actions and we can still test this component in isolation.

Reply:
I see your point. I agree with you integration tests that would just test that a button is on the page would not make sense. My point was to test that e.g. when we click on `copyWidgets` that we actually see the widget copied on the dashboard in the store.

Reply:
What do you think about doing this as an e2e test instead? I was thinking after we add playwright we could create a test which covers a bunch of these features. Thinking of a test like - drag a widget onto the dashboard, copy / paste with context menu, drag move it over, and delete with keyboard shortcut. That would cover gestures and actions in 1 shot.

Reply:
I love that idea.